### ISSUE TITLE
Links in headings are visually unrecognizable

### ISSUE THREAD

**Issue by not-my-profile** [whatwg/whatwg.org#425]
For example the heading of [section 13.2.6.5 of the HTML spec](https://html.spec.whatwg.org/multipage/parsing.html#parsing-main-inforeign) is rendered as:

![image](https://github.com/whatwg/whatwg.org/assets/73739153/e48087b8-afef-480a-a021-7582ae0abcf6)

what you cannot see is that "foreign content" is hyperlinked.

**Comment by annevk** [comment]
That's a definition, but valid point.

I wonder if we should solve the styling issue or instead attempt to move definitions out of headings as generally that seems like bad practice to me.

**Comment by domenic** [comment]
I think definitions in headings is fine, and the fact that it's not obvious that they're definitions is also fine? The point of definitions is to have something to link to; they don't need to be visually obvious, IMO.

**Comment by not-my-profile** [comment]
The point is that `html-dfn.js` turns these definitions into interactive elements and interactive elements should be visually recognizable.

Moving these definitions out of the headings could be one possible way of addressing this, and could probably also address whatwg/html#9811.

**Comment by domenic** [comment]
Yes, there are a number of interactable elements in HTML which are not properly marked up. It'd be nice to add some stuff to our build tools to emulate what Bikeshed (the other major spec build tool) does to them, which is add a little floating `#` permalink indicator.

**Comment by not-my-profile** [comment]
I don't know what you mean by "little floating `#` permalink indicator.". The bikeshed whatwg specs use the same styling as the HTML spec.

**Comment by annevk** [comment]
They have different markup which sometimes helps (and sometimes does not), but for this they appear to have the same problem: [https://fetch.spec.whatwg.org/#should-response-to-request-be-blocked-due-to-mime-type?](https://fetch.spec.whatwg.org/#should-response-to-request-be-blocked-due-to-mime-type?)

**Comment by janbrasna** [comment]
A quick PoC adding subtle 30% grey dotted underline:

```css
body.dfnEnabled dfn { text-decoration: underline dotted var(--a-underline); }
```

> ![Screen Shot 2024-07-22 at 18 34 51](https://github.com/user-attachments/assets/f02e9fc4-b4ee-44ed-8179-003ca907a331)

doesn't make it _too_ busy, but admittedly the _dfns_ are a plenty, so it may seem like a noise somewhat.
